internal/compiler/tests/syntax/functions/functions_call.slint

```diff
@@ -31,8 +42,47 @@ export Xxx := Rectangle {
         comp.f1();
 //           ^error{The function 'f1' is private. Annotate it with 'public' to make it accessible from other components}
         comp.f2();
+
+        comp.f3();
+//           ^error{The function 'f3' is protected}
     }
 
     callback xx <=> foo;
 //  ^error{Binding to callback 'xx' must bind to another callback}
 }
+
+
+export component DerComp inherits Comp {
+    public function f4() {
+        root.f1();
+//           ^error{The function 'f1' is private. Annotate it with 'public' to make it accessible from other components}
+        root.f2();
+        root.f3();
+
+        self.f1();
+//           ^error{The function 'f1' is private. Annotate it with 'public' to make it accessible from other components}
+        self.f2();
+        self.f3();
+    }
+}
+
+export component DerDerComp inherits DerComp {
+    public function f5() {
+        root.f1();
+//           ^error{The function 'f1' is private. Annotate it with 'public' to make it accessible from other components}
+        root.f2();
+        root.f3();
+//           ^error{The function 'f3' is protected}
+        self.f1();
+//           ^error{The function 'f1' is private. Annotate it with 'public' to make it accessible from other components}
+        self.f2();
+        self.f3();
+//           ^error{The function 'f3' is protected}
+
+        Glob.g1();
+//           ^warning{The function 'g1' is private. Annotate it with 'public' to make it accessible from other components. Note: this used to be allowed in previous version, but this should be considered an error}
```

Comment:
Yeah, calling private function in global used to be allowed :-/  Now i made it a warning